src/test/scala/query/AssignmentSpec.scala

```diff
@@ -0,0 +1,80 @@
+package com.originate.scalypher.test.query
+
+import com.originate.scalypher.types.ReferenceableMap
+import com.originate.scalypher.Property
+import com.originate.scalypher.Label
+import com.originate.scalypher.path.AnyNode
+import com.originate.scalypher.path.AnyRelationship
+import com.originate.scalypher.RemovePropertyAssignment
+import com.originate.scalypher.MergePropertiesAssignment
+import com.originate.scalypher.OverwriteAssignment
+import com.originate.scalypher.PropertyAssignment
+
+import org.scalatest._
+
+class AssignmentSpec extends WordSpec with Matchers {
+
+  val startNode = AnyNode()
+  val endNode = AnyNode()
+  val relationship = AnyRelationship()
+
+  val label1 = Label("label1")
+  val label2 = Label("label2")
+  val properties = Seq(
+    Property("name", "matt"),
+    Property("age", 12)
+  )
+
+  val map: ReferenceableMap = Map(
+    startNode -> "a1",
+    endNode -> "a2",
+    relationship -> "a3"
+  )
+
+  "removing properties" must {
+
+    "set elements to null" in {
+      val assignment = startNode.property("name").assignNull
+      assignment.toQuery(map) shouldBe "a1.name = NULL"
+    }
+
+  }
+
+  "assigning properties" must {
+
+    "set elements to strings" in {
+      val assignment = startNode.property("name") assign "asdf"
```

Comment:
i was considering making `:=` as an alias for `assign`. the name is arbitrary, but i didn't want to conflate the term i use here with the idea of actually assigning a value to an object. to explain a bit more... 

it's a weird situation. we aren't actually assigning "asdf" to `startNode.name` with this expression, we are creating an assignment expression that does that. so i wanted to avoid things that would normally be read as "actually assigning a value to the thing". 

does that make sense at all?